plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/AwsToolkitExplorerFactory.kt

```diff
@@ -172,6 +169,28 @@ class AwsToolkitExplorerFactory : ToolWindowFactory, DumbAware {
         }
     }
 
+    private fun inspectExistingConnection(project: Project): Boolean =
+        ToolkitConnectionManager.getInstance(project).let {
+            val conn = it.activeConnection()
+            val hasIdCRoleAccess = if (conn is AwsBearerTokenConnection) {
+                conn.scopes.contains(IDENTITY_CENTER_ROLE_ACCESS_SCOPE)
+            } else {
+                false
+            }
+
+            val isCodecatalystConn = it.activeConnectionForFeature(CodeCatalystConnection.getInstance()) != null
+            val hasIamCredential = CredentialManager.getInstance().getCredentialIdentifiers().isNotEmpty()
+
+            LOG.debug {
+                "checking existing connection(s)... " +
+                    "hasIdCPermission=$hasIdCRoleAccess; codecatalyst=$isCodecatalystConn; IAM=$hasIamCredential"
+            }
+
+            it.activeConnectionForFeature(CodeCatalystConnection.getInstance()) != null ||
+                hasIdCRoleAccess ||
+                CredentialManager.getInstance().getCredentialIdentifiers().isNotEmpty()
```

Comment:
Instead of the or at the bottom, could probably be restructured
if (CredentialManager.getInstance().getCredentialIdentifiers().isNotEmpty()) return true
//similar check for sso
//check for CC